backpack/core/derivatives/avgpoolnd.py

```diff
@@ -0,0 +1,179 @@
+        convnd.weight.data = avg_kernel
+
+        return convnd(mat)
+
+    def __check_jmp_out_as_pool(self, mat, jmp_as_pool, module):
+        V = mat.size(0)
+        if self.N == 1:
+            N, C_out, L_out = module.output.shape
+            assert jmp_as_pool.shape == (V * N * C_out, 1, L_out)
+        elif self.N == 2:
+            N, C_out, H_out, W_out = module.output.shape
+            assert jmp_as_pool.shape == (V * N * C_out, 1, H_out, W_out)
+        elif self.N == 3:
+            N, C_out, D_out, H_out, W_out = module.output.shape
+            assert jmp_as_pool.shape == (V * N * C_out, 1, D_out, H_out, W_out)
+        else:
+            raise ValueError("{}-dimensional Conv. is not implemented.".format(self.N))
+
+    def _jac_t_mat_prod(self, module, g_inp, g_out, mat):
+        self.check_exotic_parameters(module)
+
+        mat_as_pool = self.__make_single_channel(mat, module)
+        jmp_as_pool = self.__apply_jacobian_t_of(module, mat_as_pool)
+        self.__check_jmp_in_as_pool(mat, jmp_as_pool, module)
+
+        return self.reshape_like_input(jmp_as_pool, module)
+
+    def __apply_jacobian_t_of(self, module, mat):
+        C_for_conv_t = 1
+
+        convnd_t = self.convt(
+            in_channels=C_for_conv_t,
+            out_channels=C_for_conv_t,
+            kernel_size=module.kernel_size,
+            stride=module.stride,
+            padding=module.padding,
+            bias=False,
+        ).to(module.input0.device)
+
+        convnd_t.weight.requires_grad = False
+        avg_kernel = torch.ones_like(convnd_t.weight) / convnd_t.weight.numel()
+        convnd_t.weight.data = avg_kernel
+
+        V_N_C_in = mat.size(0)
+        if self.N == 1:
+            _, _, L_in = module.input0.size()
+            output_size = (V_N_C_in, C_for_conv_t, L_in)
+        elif self.N == 2:
+            _, _, H_in, W_in = module.input0.size()
+            output_size = (V_N_C_in, C_for_conv_t, H_in, W_in)
+        elif self.N == 3:
+            _, _, D_in, H_in, W_in = module.input0.size()
+            output_size = (V_N_C_in, C_for_conv_t, D_in, H_in, W_in)
+        else:
+            raise ValueError("{}-dimensional Conv. is not implemented.".format(self.N))
+
+        return convnd_t(mat, output_size=output_size)
+
+    def __check_jmp_in_as_pool(self, mat, jmp_as_pool, module):
+        V = mat.size(0)
+        if self.N == 1:
+            N, C_in, L_in = module.input0.size()
+            assert jmp_as_pool.shape == (V * N * C_in, 1, L_in)
+        elif self.N == 2:
+            N, C_in, H_in, W_in = module.input0.size()
+            assert jmp_as_pool.shape == (V * N * C_in, 1, H_in, W_in)
+        elif self.N == 3:
+            N, C_in, D_in, H_in, W_in = module.input0.size()
+            assert jmp_as_pool.shape == (V * N * C_in, 1, D_in, H_in, W_in)
+        else:
+            raise ValueError("{}-dimensional Conv. is not implemented.".format(self.N))
```

Comment:
I think we can drop that statement, since the proper range of `self.N` is already checked in the constructor